add pom.xml

psasha007 opened the pull request:
Sprint_7, он же Sprint_3

psasha007 committed:
add pom.xml

psasha007 committed:
Добавил тесты на ручку «Создать курьера». CourierCreateTest
Добавил тесты на ручку «Логин курьера». CourierLoginTest
Добавил тесты на ручку «Создать заказ». OrdersCreateTest
Добавил тесты на ручку, которая получает список заказов. OrderListTest
Добавил тесты на ручку «Удалить курьера». CourierDeleteTest
Добавил тесты на ручку «Принять заказ». OrdersToAcceptTest
Добавил тесты на ручку «Получить заказ по его номеру». OrdersGetTest
Добавил отчёт с помощью Allure.

psasha007 committed:
Переименовал на Sprint_3

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,204 @@
+    //@After
+    public void setDown(){
+        CourierLogin courierLogin = new CourierLogin("petya_ninja65", "12345");
+        CourierLoginResponseJson courierJson = given()
```
нужно удалять созданный в тесте объект, а не создавать его в методе, а затем удалять (как реализовано у тебя).

Это нужно обязательно исправить.

psasha007 replied:
сделал удаление через @After

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,204 @@
+    public void verifyPostMethodCreateNewCourier() {
+        CourierCreate courierCreate = new CourierCreate(login, password, firstName);
+        CourierCreateResponseJson courierCreateResponseJson =
+        given()
```
нужно улучшить архитектуру кода.

Сейчас в каждом тесте ты дублируешь код по формированию и отправления запроса.

Дублирование кода нужно избегать. Как это сделать:

Создать классы CourierApi и OrderApi.

В каждом классе создать методы для формирования запроса (например: создание courier, удаление courier, логин и т.д.).

В этом случае, если изменится запрос для создания курьера, тебе придется вносить правки в одном месте (в методе), а не во все тесты сразу.

Это нужно обязательно исправить.

psasha007 replied:
Попробовал улучшить архитектуру кода

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,204 @@
+        RestAssured.baseURI = BASE_URL;
+    }
+
+    //@After
```
это лучше вынести в @after.

всю чистку кода лучше выносить в @after. Сейчас если тест упадет, то чистка не будет производиться. Если удаление вынести в @after -- то очистка будет производиться всегда, даже если тест упадет.


Это нужно обязательно исправить.

psasha007 replied:
вынес в @after

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,204 @@
+    }
+
+    @Parameterized.Parameters
+    public static Object[] getSumData()
```
Для этих сценариев параметризация не уместна.

psasha007 replied:
убрал параметризацию из этого теста

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,204 @@
+                .body(courierCreate2) // передача объекта с данными
+                .when()
+                .post(REQUEST_CREATE_NEW_COURIER) // отправка POST-запроса
+                .then().assertThat().statusCode(STATUS_CODE_409);
```
статус коды можно было взять из библиотеки:
import  static  org.apache.http.HttpStatus.*;

statusCode(SC_CREATED)

psasha007 replied:
статус коды взял из библиотеки:
import static org.apache.http.HttpStatus.*;
и добавил в код.

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,204 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = BASE_URL;
```
это можно было реализовать через базовый класс, в котором задать  RequestSpecification. От этого базового класса потом наследовать классы для формирования запросов для Курьера и Заказа.

psasha007 replied:
Попробовал реализовать через базовый класс

psasha007 committed:
- Поправил - не создавать объект в методе.
- Улучшил архитектуру кода (теперь не дублирую код по формированию и отправлению запроса, создал Api классы).
- Вынес удаление в @after
- В классе CourierCreateTest.java убрал параметризацию
- Теперь статус коды беру из библиотеки:
         import static org.apache.http.HttpStatus.*;
- Попробовал реализовать через базовый класс для RequestSpecification

alesyaisachenko commented on src/main/java/CourierApi.java:
```diff
@@ -0,0 +1,37 @@
+import static org.apache.http.HttpStatus.SC_OK;
+
+public class CourierApi extends BaseApi {
+    public void courierCreateStatusCode(CourierCreate courierCreate, int statusCode){
```
используй аннотацию [@Step](https://github.com/step), для того чтобы отчеты в алюр были более информативные.

alesyaisachenko commented on src/main/java/CourierApi.java:
```diff
@@ -0,0 +1,37 @@
+
+public class CourierApi extends BaseApi {
+    public void courierCreateStatusCode(CourierCreate courierCreate, int statusCode){
+         given()
```
лучше создать один метод по отправке запроса и все. Проверять ответ уже в самом тесте и не выделять для него отдельный метод, то есть не стоило выделать отдельны методы courierCreateResponseJson(CourierCreate courierCreate), login(CourierCredentials cred).

Коммент относится ко всему коду.
